### PR TITLE
OHRM5X-8:Implement menu items for oxd-layout

### PR DESCRIPTION
**Pull request by RajithaKumara**

### CODE REVIEW

File: dbscript/dbscript-2.sql [hunk 1]
```diff
@@ -3531,3 +3531,9 @@ INSERT INTO `ohrm_i18n_group` (`name`, `title`) VALUES
 ('buzz', 'Buzz'),
 ('marketplace', 'Marketplace'),
 ('mobile', 'Mobile');
+
+-- ------------------------------
+--        OrangeHRM 5.x
+-- ------------------------------
+
+ALTER TABLE `hs_hr_config` CHANGE `key` `name` VARCHAR(100);
```
Comment (samanthajayasinghe) [inline dbscript/dbscript-2.sql:3539]: Are we going to disabled all non-implemented  screens when install ?

File: symfony/plugins/orangehrmCorePlugin/Authorization/Dao/DataGroupDao.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+    /**
+     * @param string[]|string $dataGroups
+     * @param int $userRoleId
+     * @param bool $selfPermission
+     * @return DataGroupPermission[]
+     * @throws DaoException
+     */
+    public function getDataGroupPermission($dataGroups, int $userRoleId, bool $selfPermission = false): array
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmCorePlugin/Authorization/Dao/DataGroupDao.php:56]: missing type for `dataGroups`

File: symfony/plugins/orangehrmCorePlugin/Authorization/UserRole/EssUserRole.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+namespace OrangeHRM\Core\Authorization\UserRole;
+
+/**
+ * Description of EssUserRole
+ *
+ * @author samith
+ */
+class EssUserRole extends AbstractUserRole {
+
+    public function getAccessibleEmployeeIds($operation = null, $returnType = null, $requiredPermissions = []) {
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmCorePlugin/Authorization/UserRole/EssUserRole.php:29]: missing return types

File: symfony/plugins/orangehrmCorePlugin/Dao/BaseDao.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+{
+    /**
+     * @param string $entityClass
+     * @return ObjectRepository|EntityRepository
+     *
+     * @template T
+     * @psalm-param class-string<T> $entityClass
+     * @psalm-return EntityRepository<T>
+     */
+    protected function getRepository(string $entityClass)
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmCorePlugin/Dao/BaseDao.php:40]: missing return types

File: symfony/plugins/orangehrmCorePlugin/Dao/ConfigDao.php [hunk 5]
```diff
@@ -18,84 +17,79 @@
...
+use OrangeHRM\Entity\Config;
+use OrangeHRM\Framework\Logger;
+
 /**
  * Config Dao: Manages configuration entries in hs_hr_config
  *
  */
-class ConfigDao extends BaseDao {
-
-    private $logger;
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmCorePlugin/Dao/ConfigDao.php:27]: private $logger = null ;

File: symfony/plugins/orangehrmCorePlugin/Helper/MenuHelper.php [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+class MenuHelper
+{
+    /**
+     * @param string $baseUrl
+     * @return array
+     * @throws DaoException
+     * @throws ServiceException
+     */
+    public static function getMenuItems(string $baseUrl): array
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmCorePlugin/Helper/MenuHelper.php:35]: `getMenuItems(string $baseUrl)` not make much sense .. can't we inject the baseurl in other way?

File: dbscript/dbscript-2.sql (hunk 1 above)
Comment (RajithaKumara) [reply dbscript/dbscript-2.sql:3539]: Done

Comment (RajithaKumara) [reply dbscript/dbscript-2.sql:3539]: Done

File: symfony/plugins/orangehrmCorePlugin/Dao/BaseDao.php (hunk 4 above)
Comment (RajithaKumara) [reply symfony/plugins/orangehrmCorePlugin/Dao/BaseDao.php:40]: here also Doctrine return `ObjectRepository|EntityRepository`

File: symfony/plugins/orangehrmCorePlugin/Dao/ConfigDao.php (hunk 5 above)
Comment (RajithaKumara) [reply symfony/plugins/orangehrmCorePlugin/Dao/ConfigDao.php:27]: this logger property removed

File: symfony/plugins/orangehrmCorePlugin/Authorization/UserRole/EssUserRole.php (hunk 3 above)
Comment (RajithaKumara) [reply symfony/plugins/orangehrmCorePlugin/Authorization/UserRole/EssUserRole.php:29]: All the User Role classes have to properly refactor.

File: symfony/plugins/orangehrmCorePlugin/Authorization/Dao/DataGroupDao.php (hunk 2 above)
Comment (samanthajayasinghe) [reply symfony/plugins/orangehrmCorePlugin/Authorization/Dao/DataGroupDao.php:56]: I think it always should be array

**Pull request merged**